locator/tablets.cc

```diff
@@ -326,6 +328,7 @@ tablet_transition_stage tablet_transition_stage_from_string(const sstring& name)
 // The names are persisted in system tables so should not be changed.
 static const std::unordered_map<tablet_transition_kind, sstring> tablet_transition_kind_to_name = {
         {tablet_transition_kind::migration, "migration"},
+        {tablet_transition_kind::intranode_migration, "intranode_migration"},
```

Comment:
On the other hand, it's not a big deal. We can't pretend that sharding is an implementation detail we can hide.

Reply:
It could be computed from replica sets, but as mentioned in the cover commit log, this information is needed on the hot path so it's better to have it in the transition kind.

Reply:
Yes, but we could have kept the internal representation separate.

In any case, it's okay to have it exposed. And maybe better too, so it stands out when there are problems, and as I said sharding is not an implementation detail (we keep the shard numbers in the replica set).